doc/manual/src/glossary.md

```diff
@@ -208,44 +208,85 @@
 
-    If the output of a compiler is to be run in a [derivation], the compiler's [target system] must be executable by the [derivation system](language/derivations.md#attr-system).
+    Programs that run natively in a [derivation] must have a [build configuration] with a [host platform] that can by executed by the [derivation system](language/derivations.md#attr-system).
+
+    This term is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+
+  - [target platform]{#gloss-target-platform}
+
+    [target platform]: #gloss-target-platform
+
+    The [platform] for which a compiler will generate code, typically a dash-separated string.
+
+    When discussing a build that does not produce a compiler, we only use [build platform] and [host platform] to describe it. This also applies to compilers with a dynamic target, which is not part of the [build configuration].
+
+    We try to avoid discussing the actual build from the perspective of its compiler, instead focusing on the build itself.
+    Otherwise we won't have a term to distinguish the actual [target platform] from what we should have called the [host platform].
+
+    If the output of a compiler is to be run in a [derivation], the compiler's [target platform] must be executable by the [derivation system](language/derivations.md#attr-system).
+
+    This term is based on the [GCC system terminology](#gloss-gcc-system-terminology).
 
   - [GCC system terminology]{#gloss-gcc-system-terminology}
 
     The Nix ecosystem uses the [GCC system terminology](https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html) for the names and roles of _build_, _host_, and _target_, but may apply the term _cross_ more loosely.
 
-    See [host system], [build system] and [target system] for how each relates to [derivation].
+    See [host platform], [build platform] and [target platform] for how each relates to [derivation].
+
+  - [build configuration]{#gloss-build-configuration}
+
+    [build configuration]: #gloss-build-configuration
+
+    The software configuration whose effects enable the build process and most of which also influence the [build product](#gloss-build-product).
+
+    Build configuration does not have a single agreed upon definition, but consists of a potentially broad set of settings that include:
+    - [build platform], often implicitly
+    - [host platform]
+    - [target platform], only when building a compiler with a static target platform
+    - options passed to a compiler, such optimization settings and the host platform (when using a compiler with a dynamic target platform)
+    - dependency options, such as which build tools and dependencies to use
+    - anything specified in a custom build script
+
+    In the context of Nix, the [build configuration] can be thought of as a [derivation] minus the source code, plus intentional build impurities.
+
+  - [build product]{#gloss-build-product}
```

Comment:
```suggestion
  - [build result]{#gloss-build-product}
```

We started calling it that in the architecture docs with @Ericson2314 and it's already a term in the manual and nix.dev. It doesn't meen it have to be, but I like it because it doesn't raise mathematical associations.